### ISSUE TITLE
Trying to add a local variable at positive storage offset 4

### ISSUE THREAD

**Issue by xusheng6** [Vector35/binaryninja-api#5716]
I get the error message in the title when I open this arm64 Linux helloworld binary. The binary is compiled natively on an arm64 VM 

[helloworld.zip](https://github.com/user-attachments/files/16171675/helloworld.zip)

**Comment by negasora** [comment]
This is because the CFA offset changes over the course of the function but since we need to define all stack variables statically we just use the last CFA offset from the CIE. We _could_ heuristically determine that there are "initial" and "working" CFA offsets from parsing the CIE but I'm not sure how accurate that would be.

`llvm-dwarfdump helloworld --eh-frame` output for `main`:
```
00000088 0000001c 0000008c FDE cie=00000000 pc=000007d4...00000858
  Format:       DWARF32
  DW_CFA_advance_loc: 4
  DW_CFA_def_cfa_offset: +48
  DW_CFA_offset: W29 -48
  DW_CFA_offset: W30 -40
  DW_CFA_advance_loc: 124
  DW_CFA_restore: W30
  DW_CFA_restore: W29
  DW_CFA_def_cfa_offset: +0
  DW_CFA_nop:
  DW_CFA_nop:
  DW_CFA_nop:

  0x7d4: CFA=WSP
  0x7d8: CFA=WSP+48: W29=[CFA-48], W30=[CFA-40]
  0x854: CFA=WSP
```

**Comment by negasora** [comment]
Fixed in `4.2.6177-dev`
